app/models/api_key.rb

```diff
@@ -0,0 +1,21 @@
+class ApiKey < ApplicationRecord
+  API_SCOPES = %i[index_rubygems push_rubygem yank_rubygem add_owner remove_owner webhook_actions show_dashboard].freeze
+
+  belongs_to :user
+  validates :user, :name, :hashed_key, presence: true
+  validate :exclusive_show_dashboard_scope, if: :show_dashboard?
+
```

Comment:
Multiple API keys are able to have the same nickname, I think we should have a unique validation on `name` to avoid confusion for users and avoid accidentally deleting the wrong API key. Thoughts?

Reply:
Inforcing name uniqueness will mean the user will have to face some frustration when signing in from cli (where they can't see the existing names) and keep getting asked to enter email and password. Also, we would have to add random bits to the default key name we are suggesting in cli.

If you think these issues don't compare to what you are pointing out, we can add name uniqueness. I don't have any strong opinion about this, I was just leaning towards not making users fail their signin attempts.